commonknowledge/mapped
# Improvements to the UI for importing and enriching data [MAP-347] [MAP-78] [MAP-77]

## Description
This PR adds:
if an import or update is in a queue behind others indicate this in the UI
a progress bar when importing and updating records
when an job is going to take longer than 5 minutes tell the user it is finished via email

## Motivation and Context
Addresses issues [MAP-347,](https://linear.app/commonknowledge/issue/MAP-347/when-an-job-is-going-to-take-longer-than-it-is-reasonable-for-someone), [MAP-77](https://linear.app/commonknowledge/issue/MAP-77/no-progress-bar-or-estimated-time-on-imported-records), [MAP-78](https://linear.app/commonknowledge/issue/MAP-78/if-an-import-or-update-is-in-a-queue-behind-others-indicate-this-in)

## How Can It Be Tested?
Run the branch locally
Import and/or update a data source
**Testing progress bar:**

- If working with smaller data sets update the IMPORT_UPDATE_ALL_BATCH_SIZE to a small size e.g. 2
- Observe a progress bar during the import or update

**Testing job queue:**

- Try requesting more than one import/update job at once and observe that the UI indicates that you are in a queue

**Testing email notification:**

- set timedelta in get_scheduled_batch_job_progress to something low (e.g. 1 minute)
- observe that the UI indicates you will be sent an email
- observe the email printed to the console
